activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -64,14 +60,16 @@ def self.#{sym}
         end
       EOS
 
-      unless options[:instance_reader] == false || options[:instance_accessor] == false
+      if instance_reader && instance_accessor
         class_eval(<<-EOS, __FILE__, __LINE__ + 1)
           def #{sym}
             @@#{sym}
           end
         EOS
       end
-      class_variable_set("@@#{sym}", yield) if block_given?
+
+      default = yield if block_given? && default.nil?
```

Comment:
But actually, looking at the test, it was pretty bad to begin with. It doesn't exercise the multiple invocation. 🤦‍♂️